examples/cpp/SampleCppMini/main.cpp

```diff
@@ -88,6 +88,7 @@ void test_cpp_api(const char * token, int ticketType, const char *ticket)
     logger->LogEvent(detailed_event);
 
     // Shutdown
+    LogManager::Flush();
```

Comment:
There's really something broken in the SDK if it requires `Flush` before `FlushAndTeardown`. The intent of `FlushAndTeardown` to perform `Flush`, wait for (configurable) N seconds, then `Teardown`.

Reply:
Yes, that was what I thought too. But I don't see that's happening in FlushAndTeardown by looking into the code. Also we let this get documented here - https://github.com/microsoft/cpp_client_telemetry/blob/main/docs/use-after-free.md#flushandteardown

``
If one wishes to use this backstop to quiesce the SDK, the application should call Flush before FlushAndTeardown. This looks redundant (doesn't FlushAndTeardown actually flush, you might ask), but despite the method name FlushAndTeardown, the call to Flush is required to ensure persistence before the PauseActivity quiesce.
``

Reply:
I think this document was actually contributed at the same time when things got broken.. 🤣 I have a theory that this got broken by the zombie-logger change. My team uses a slightly older version of SDK, and we do not see the issue.